src/geode/inspector/criterion/intersections/model_intersections.cpp

```diff
@@ -94,31 +95,18 @@ namespace
         {
         }
 
-        void operator()( geode::index_t t1_id, geode::index_t t2_id )
-        {
-            if( same_surface_ && t1_id == t2_id )
-            {
-                return;
-            }
-            const auto t1_vertices = mesh1_.polygon_vertices( t1_id );
-            const auto t2_vertices = mesh2_.polygon_vertices( t2_id );
-            const auto common_vertices =
-                triangles_common_vertices( t1_vertices, t2_vertices );
-            if( common_vertices.size() == 3
-                || triangles_intersect(
-                    t1_id, t2_id, t1_vertices, t2_vertices, common_vertices ) )
-            {
-                intersecting_triangles_.emplace_back( t1_id, t2_id );
-            }
-        }
+        virtual ~ModelSurfacesIntersectionBase() = default;
+
+        virtual bool operator()(
```

Comment:
The polymorphism is not really needed. We could remove the destructor and the virtual operator I think and directly use the one from the children